### PR TITLE
[v2.8] Updated Rancher to import the framework from the repo rancher/shepherd.

### PR DESCRIPTION
**Pull request by igomez06**
## Problem
 The testing framework has been moved to its own repo https://github.com/rancher/shepherd/ so we need to import it to rancher/rancher
 
## Solution
 Removed tests/framework and imported rancher/shepherd to the repo.
 
### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

**Review by markusewalker — APPROVED** [review]
LGTM

File: go.mod [hunk 1]
```diff
@@ -167,7 +167,8 @@ require (
 require (
 	github.com/antihax/optional v1.0.0
 	github.com/containers/image/v5 v5.25.0
-	github.com/rancher/rancher/pkg/apis v0.0.0-20230915232223-a9ea4ce4a5ba
+	github.com/rancher/rancher/pkg/apis v0.0.0-20231214165302-b2b815c7402d
```
Comment (caliskanugur) [inline go.mod:170]: Just to make sure this bit, does this need an update?

**Review by caliskanugur — COMMENTED** [review]
I've only one question for this backport

**Review by Priyashetty17 — DISMISSED** [review]
lgtm

**Review by Priyashetty17 — APPROVED** [review]

**Review by markusewalker — DISMISSED** [review]

**Review by git-ival — APPROVED** [review]
Needs successful drone build but LGTM 🚀

**Review by git-ival — APPROVED** [review]

**Review by slickwarren — DISMISSED** [review]

**Review by floatingman — DISMISSED** [review]
Looks good.

**Review by caliskanugur — APPROVED** [review]
Looks good to me

**Review by slickwarren — APPROVED** [review]

**Review by markusewalker — APPROVED** [review]

**Review by anupama2501 — APPROVED** [review]

**Review by Priyashetty17 — APPROVED** [review]

**Pull request merged**